src/compiler/scala/tools/nsc/backend/opt/ConstantOptimization.scala

```diff
@@ -28,8 +28,8 @@ abstract class ConstantOptimization extends SubComponent {
   import global._
   import icodes._
   import icodes.opcodes._
-  import scala.tools.nsc.backend.icode.Primitives._
-  import scala.tools.nsc.backend.icode.Opcodes._
+  import scala.tools.nsc.backend.jvm.Primitives._
```

Comment:
why did you move them? with the same reasoning, I believe we should keep them where they are now.